From 7eba18e1c0b994180e173e9343c7fe50819d9732 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@oss.nxp.com>
Date: Tue, 17 Aug 2021 11:43:16 +0300
Subject: [PATCH 4/4] s32: Fix memory reservation for MMU tables

This commit removes the set of TLB address from platform code
because this area is not reserved and its content can be altered.
This new approach will rely on TLB settings made by reserve_mmu
function.

Upstream-Status: Pending

Issue: ALB-7553
Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@oss.nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/cpu/armv8/s32/cpu.c | 2 --
 include/configs/s32.h        | 5 -----
 2 files changed, 7 deletions(-)

diff --git a/arch/arm/cpu/armv8/s32/cpu.c b/arch/arm/cpu/armv8/s32/cpu.c
index 8edf6d9a8d..ed827e86d1 100644
--- a/arch/arm/cpu/armv8/s32/cpu.c
+++ b/arch/arm/cpu/armv8/s32/cpu.c
@@ -249,9 +249,7 @@ static inline void final_mmu_setup(void)
 	mem_map = final_map;
 
 	/* global data is already setup, no allocation yet */
-	gd->arch.tlb_addr = S32_SDRAM_MMU_TABLES_BASE;
 	gd->arch.tlb_fillptr = gd->arch.tlb_addr;
-	gd->arch.tlb_size = get_tlb_size();
 
 #if defined(CONFIG_S32_SKIP_RELOC) && !defined(CONFIG_S32_ATF_BOOT_FLOW)
 	sram_clr(gd->arch.tlb_addr, gd->arch.tlb_size);
diff --git a/include/configs/s32.h b/include/configs/s32.h
index 330d6ba5c4..866ec7c6e9 100644
--- a/include/configs/s32.h
+++ b/include/configs/s32.h
@@ -560,11 +560,6 @@
 
 #define S32_MMU_TABLES_OFFSET		CONFIG_SYS_INIT_SP_OFFSET
 #define S32_IRAM_MMU_TABLES_BASE	(IRAM_BASE_ADDR + S32_MMU_TABLES_OFFSET)
-#ifdef CONFIG_S32_SKIP_RELOC
-#define S32_SDRAM_MMU_TABLES_BASE	S32_IRAM_MMU_TABLES_BASE
-#else
-#define S32_SDRAM_MMU_TABLES_BASE	(CPU_RELEASE_ADDR + PGTABLE_SIZE)
-#endif
 
 #define CONFIG_SYS_MMC_ENV_DEV		0
 #define MMC_PART_FAT			1
-- 
2.17.1

